### ISSUE TITLE
l3keys: properties `.cs_set...:N` in addition to `.cs_set...:Np` ones?

### ISSUE THREAD

**Issue by muzimuzhi** [latex3/latex3#1507]
Corresponds to `\cs_set...:Npn` functions, `l3keys` provides key properties `.cs_set:Np`, `.cs_set_protected:Np`, and friends.

For completeness, maybe properties like `.cs_set:N` and `.cs_set_protected:N` can also be provided, which correspond to `\cs_set:Nn` and friends.

**Comment by josephwright** [comment]
It seems pretty unlikely that a key name would contain the relevant arg. spec. to use `.cs_set:N` - feels like it would be at the wrong level.

**Comment by Skillmon** [comment]
I don't think this is about the key-name, but about parsing the cs which is the handle's argument. So:

```
\keys_define:nn { foo } { bar .cs_set:N = \__foo_bar:nn }
```

should be identical to

```
\keys_define:nn { foo } { bar .cs_set:Np = \__foo_bar:nn #1#2 }
```

**Comment by muzimuzhi** [comment]
As @Skillmon explained, `.cs_set:Np` shares the same issue with `.cs_set:N` that key name may not contain arg. spec..

Meanwhile, if a user wants to stick to `\cs_set:Nn` family instead of `\cs_set:Npn` family, then `.cs_set:N` would be needed too.
